client.go

```diff
@@ -180,31 +183,50 @@ func runStreamedCommand(img, cmd string, getError bool) error {
 // It returns the ID of the container spawned from the image.
 // Note well: the container is NOT deleted when the given command terminates.
 func runCommandInContainer(img string, cmd []string, dst io.Writer) (string, error) {
-	return startContainer(img, cmd, true, dst)
+	id, err := createContainer(img, cmd)
+	if err != nil {
+		log.Println(err)
+		return "", err
+	}
+
+	execute := true
+	for execute {
+		id, err = startContainer(id, cmd, true, dst)
+		switch err.(type) {
+		case dockerError:
+			de := err.(dockerError)
+			if isZypperExitCodeSevere(de.exitCode) {
+				return "", err
+			}
+			if de.exitCode == zypperExitInfRestartNeeded {
+				execute = true
+			}
+		default:
+			execute = false
+			return id, err
+		}
+		return id, err
```

Comment:
Doesn't this line mean the loop will never run more than once?

Reply:
which one? the "return id, err" ?

Reply:
Line 208. To me it looks like:

```
for someCond {
  // some code
  return blah
}
```

Which will always return `blah` (you need to use `continue` or similar to make the loop not hit that `return` statement).